### ISSUE TITLE
V0.16.0 and v0.17.0 don't work in K8s 1.10

### ISSUE THREAD

**Issue by derrickburns** [solo-io/gloo#828]
Try doing a clean install.  Complains of `spec.version` not being set for settings crd.

```
"Mon, 15 Jul 2019 05:16:21 UTC: github.com/solo-io/gloo/projects/gloo/cmd/main.go:20"	err in main: "creating base Settings resource client: failed to register crd: CustomResourceDefinition.apiextensions.k8s.io \"settings.gloo.solo.io\" is invalid: spec.version: Required value"
```

**Comment by yuval-k** [comment]
when checking I see 'spec.version' field should be set in our CRD; A few questions to help us reproduce this issue:
- Is this a new kubernetes cluster? Or did this cluster had gloo installed before?
- How did you install gloo? (via helm? glooctl? kubectl?)

**Comment by kdorosh** [comment]
Which version of kubernetes are you running? (both client/server)

**Comment by derrickburns** [comment]
This is locally with using the local K8s with Docker:

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-20T04:49:16Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by derrickburns** [comment]
I installed gloo with helm as a dependency.  

See

https://github.com/tidepool-org/development/tree/k8s/charts/tidepool/0.1.5

**Comment by kdorosh** [comment]
We were able to reproduce the issue using versions 0.16.0 and 0.17.0 of gloo with a kubernetes server version 1.10.0. Upgrading to 1.11.0 (or higher) resolved the issue for us.

@derrickburns is upgrading a possible resolution path for you? We are still digging into the root cause/determining the difficulty of supporting older versions of kubernetes.

Reproduce steps, for those curious:
* download 0.16.0/0.17.0 version of `glooctl`
* `minikube delete`
* `minikube start --kubernetes-version=v1.10.0`
* `glooctl install gateway`
The pods in the gloo-system namespace will fail to come up with the linked error message.

**Comment by derrickburns** [comment]
I was trying to get something working with the version of k8s that comes with Docker for Mac.

Derrick

**Comment by ilevine** [comment]
the reason this was not caught is because our CI uses kube 1.11 for testing; we can't currently use 1.10 as it is not present in GKE; We are forming a plan to test older versions in CI.

**Comment by joekelley** [comment]
Hi @derrickburns, this should be resolved in the next release of gloo (v0.17.4).
